third_party/java/arcs/build_defs/internal/schemas.bzl

```diff
@@ -100,3 +100,38 @@ def arcs_kt_schema(name, srcs, deps = [], package = "arcs.sdk"):
         platforms = ["jvm", "wasm"],
         deps = ARCS_SDK_DEPS,
     )
+
+def _proto2schema_impl(ctx):
+    output_name = ctx.label.name + ".kt"
+    out = ctx.actions.declare_file(output_name)
+
+    args = ctx.actions.args()
+
+    args.add_all("--outfile", [output_name])
+    args.add_all("--outdir", [out.dirname])
+    args.add_all("--package-name", [ctx.attr.package])
+    args.add_all([src.path for src in ctx.files.srcs])
+
+    ctx.actions.run(
+        inputs = ctx.files.srcs,
+        outputs = [out],
+        arguments = [args],
+        executable = ctx.executable._compiler,
+    )
+
+    return [DefaultInfo(files = depset([out]))]
+
+proto2schema = rule(
```

Comment:
why proto2schema? What is the ultimate output of this rule? As far as I understand ProdEx it should be Plans, not schemas. I may however have some wrong ideas. @csilvestrini ?

Reply:
This is a different story to recipe2plan. I'll chat you the ticket.

Reply:
@mmandlis can speak to this: the most immediate need with codegeneration is creating schemas (and `Type`s) from manifests.

Reply:
I wonder if consolidating the schema and plan generation into a single rule would be a good idea? Are we likely to want to do them separately?